kcidb/test_oo.py

```diff
@@ -203,3 +203,240 @@ def test_two_object(self):
+                        "id": "kernelci:valid2_1",
+                        "build_id": "kernelci:valid2",
+                        "origin": "kernelci", "status": "PASS", "path": "pass1"
+                    },
+                    {
+                        "id": "kernelci:valid2_2",
+                        "build_id": "kernelci:valid2",
+                        "origin": "kernelci", "status": "PASS", "path": "pass2"
+                    },
+                    {
+                        "id": "kernelci:valid2_3",
+                        "build_id": "kernelci:valid2",
+                        "origin": "kernelci", "status": "FAIL", "path": "fail"
+                    },
+                    {
+                        "id": "redhat:valid1_1",
+                        "build_id": "redhat:valid1",
+                        "origin": "redhat", "status": "PASS", "path": "pass1"
+                    },
+                    {
+                        "id": "redhat:valid1_2",
+                        "build_id": "redhat:valid1",
+                        "origin": "redhat", "status": "PASS", "path": "pass2",
+                    },
+                    {
+                        "id": "redhat:valid1_3",
+                        "build_id": "redhat:valid1",
+                        "origin": "redhat", "status": "FAIL", "path": "fail",
+                    },
+                    {
+                        "id": "redhat:valid2_1",
+                        "build_id": "redhat:valid2",
+                        "origin": "redhat", "status": "PASS", "path": "pass1"
+                    },
+                    {
+                        "id": "redhat:valid2_2",
+                        "build_id": "redhat:valid2",
+                        "origin": "redhat", "status": "PASS", "path": "pass2",
+                    },
+                    {
+                        "id": "redhat:valid2_3",
+                        "build_id": "redhat:valid2",
+                        "origin": "redhat", "status": "FAIL", "path": "fail",
+                    },
+                ]
+            }
+        )
+        self.client = kcidb.oo.Client(source)
+
+    def test_traversing_revision_links(self):
+        """Check that revision's links are successfully traversed."""
+
+        revision = self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", ""]#'
+        )['revision'][0]
+
+        self.assertListContains(revision.checkouts, 3, "checkout")
+        self.assertListContains(revision.builds, 6, "build")
+        self.assertListContains(revision.tests, 12, "test")
+
+        self.assertFalse(revision.checkouts_valid)
+        self.assertFalse(revision.builds_valid)
+
+        self.assertIsInstance(revision.tests_root, kcidb.oo.Node)
+
+        self.assertDictContains(revision.repo_branch_checkouts, {
+            "https://repo_valid": {"valid1": 1, "valid2": 1},
+            "https://repo_invalid": {"invalid": 1}
+        }, type_name="checkout")
```

Comment:
With the function above this would be something like this:
```python
        self.assertContains(repo.revision_branch_checkout, {
            "https://repo_valid": {
                "valid1": (kcidb.oo.Checkout, 1),
                "valid2": (kcidb.oo.Checkout, 1),
            },
            "https://repo_invalid": {
                "invalid": (kcidb.oo.Checkout, 1),
            },
        })
```